packages/playground/cli/src/worker-thread.ts

```diff
@@ -0,0 +1,178 @@
+import type { PHP, SupportedPHPVersion } from '@php-wasm/universal';
+import { PHPWorker, consumeAPI, exposeAPI } from '@php-wasm/universal';
+import type { FileLockManager } from '@php-wasm/node';
+import { createNodeFsMountHandler, loadNodeRuntime } from '@php-wasm/node';
+import { EmscriptenDownloadMonitor } from '@php-wasm/progress';
+import { parentPort } from 'worker_threads';
+import { bootWordPress } from '@wp-playground/wordpress';
+import { sprintf } from '@php-wasm/util';
+import { rootCertificates } from 'tls';
+
+export interface Mount {
+	hostPath: string;
+	vfsPath: string;
+}
+
+export type PrimaryWorkerBootOptions = {
+	wpVersion?: string;
+	phpVersion?: SupportedPHPVersion;
+	absoluteUrl: string;
+	mountsBeforeWpInstall: Array<Mount>;
+	mountsAfterWpInstall: Array<Mount>;
+	wordPressZip?: ArrayBuffer;
+	sqliteIntegrationPluginZip?: ArrayBuffer;
+	firstProcessId: number;
+	processIdSpaceLength: number;
+	dataSqlPath?: string;
+	followSymlinks: boolean;
+	trace: boolean;
+};
+
+function mountResources(php: PHP, mounts: Mount[]) {
+	for (const mount of mounts) {
+		php.mkdir(mount.vfsPath);
+		php.mount(mount.vfsPath, createNodeFsMountHandler(mount.hostPath));
+	}
+}
+
+/**
+ * Print trace messages from PHP-WASM.
+ *
+ * @param {number} processId - The process ID.
+ * @param {string} format - The format string.
+ * @param {...any} args - The arguments.
+ */
+function tracePhpWasm(processId: number, format: string, ...args: any[]) {
+	// eslint-disable-next-line no-console
+	console.log(
+		performance.now().toFixed(6).padStart(15, '0'),
+		processId.toString().padStart(16, '0'),
+		sprintf(format, ...args)
+	);
+}
+
+export class PlaygroundCliWorker extends PHPWorker {
+	booted = false;
+
+	/**
+	 * A string representing the requested version of WordPress.
+	 */
+	requestedWordPressVersion: string | undefined;
+
+	/**
+	 * A string representing the version of WordPress that was loaded.
+	 */
+	loadedWordPressVersion: string | undefined;
+
+	unmounts: Record<string, () => any> = {};
+
+	constructor(monitor: EmscriptenDownloadMonitor) {
+		super(undefined, monitor);
+	}
+
+	async boot({
+		absoluteUrl,
+		mountsBeforeWpInstall,
+		mountsAfterWpInstall,
+		phpVersion = '8.0',
+		wordPressZip,
+		sqliteIntegrationPluginZip,
+		firstProcessId,
+		processIdSpaceLength,
+		dataSqlPath,
+		followSymlinks,
+		trace,
+	}: PrimaryWorkerBootOptions) {
+		if (this.booted) {
+			throw new Error('Playground already booted');
+		}
+		this.booted = true;
+
+		let nextProcessId = firstProcessId;
+		const lastProcessId = firstProcessId + processIdSpaceLength - 1;
+		const fileLockManager = consumeAPI<FileLockManager>(parentPort!);
+		await fileLockManager.isConnected();
+
+		try {
+			const constants: Record<string, string | number | boolean | null> =
+				{
+					WP_DEBUG: true,
+					WP_DEBUG_LOG: true,
+					WP_DEBUG_DISPLAY: false,
+				};
+
+			const requestHandler = await bootWordPress({
```

Comment:
Should this be just bootRequestHandler now since WP is already installed?

Reply:
> Should this be just bootRequestHandler now since WP is already installed?

Added this to the follow up issue #2293